### DISCUSSION TITLE
Done my build. NanoELS + DRO

### DISCUSSION THREAD

**Discussion by vegetate7** [kachurovskiy/nanoels#181] [Show and tell]
So after long time my build done.
![IMG_20240219_210551](https://github.com/kachurovskiy/nanoels/assets/57148620/7e7cb367-2440-45db-930b-b561af337b62)

Apron is custom built form 2 piece of laser cut 6mm steel, pile of gears, pulleys, bearings and 3dprinted parts. Main goal was to make ballscrew conversion and  same time to keep manual feed flywheel functional in it's legacy place. I decided to use a fixed screw and rotating nut construction.
Sorry, didn't take puctures before assembly, some CAD screenshots to get an idea:
![cnc_apron_v9_right](https://github.com/kachurovskiy/nanoels/assets/57148620/9932c4a0-e790-467a-9d34-ceb5d6d1b29b)
![cnc_apron_v9_right_cutout](https://github.com/kachurovskiy/nanoels/assets/57148620/ed7c87b0-d37e-4a51-832b-a46a10e1bfb8)

Assembled
![IMG_20240219_210532](https://github.com/kachurovskiy/nanoels/assets/57148620/686088eb-c99f-47d6-a713-1256bfb2b1c9)
![IMG_20240219_210541](https://github.com/kachurovskiy/nanoels/assets/57148620/d8390850-e2aa-4b04-964c-5f3ad3285b0d)

DRO with sensor cables splitters
![IMG_20240219_210608](https://github.com/kachurovskiy/nanoels/assets/57148620/e50d38a5-a33f-455f-95c6-d083b1bfab6e)

NanoELS mount box (3d printed)
![IMG_20240219_210616](https://github.com/kachurovskiy/nanoels/assets/57148620/54cf2e2d-84d3-41b2-b8bd-3cbd546472d0)

Power box
![IMG_20240219_210630](https://github.com/kachurovskiy/nanoels/assets/57148620/a66e8c28-b688-4eac-ac59-294a2dc91645)
![IMG_20240219_210637](https://github.com/kachurovskiy/nanoels/assets/57148620/48606714-686b-4fa9-bdf9-2050fe0ce079)
![IMG_20240219_210654](https://github.com/kachurovskiy/nanoels/assets/57148620/0a47b1c2-ad49-4fc4-b86d-3f516ce20d5c)

Actually, the DRO part of firmware does only 2 tasks: sync axis position value to DRO value while stepper disabled, and take out backlash when starting first move.
Small video of  combined manual/auto movement, stop set, go to stop.
https://youtube.com/shorts/L_4iHvxltL0?feature=share

Previuos demo (before ballscrew HW were installed)
https://youtu.be/wIaeWsB7uOQ

**Comment by kachurovskiy** [comment]
This looks really cool! But I won't pretend like I understand what's going on 😄 Congratulations 👏🎉

**Reply by vegetate7** [reply]
It's hard for me to look on phone screen, then on buttons, then back to screen :) May be some time I'll ask my wife to shot a video.
WRT first video: I zero axis Z, then set left stop at 10mm by keypad, then manually move carriage and then use "left" button to move to the stop. 
There some strange behavior tho, I think it's because NEEDS_REST enabled. 
Movements is much more solid while in any of auto modes.

**Comment by kachurovskiy** [comment]
@vegetate7 can you please share the model files for the H4 mounting base? Many thanks!

**Comment by vegetate7** [comment]
[nanoels_box.zip](https://github.com/kachurovskiy/nanoels/files/14762675/nanoels_box.zip)
![IMG_20240326_235325](https://github.com/kachurovskiy/nanoels/assets/57148620/27bca3a3-3d32-428d-877a-a1877091c806)

Ok. Tho I do not think it will be usefull. It's very specific for my setup: DB9 connectors for DRO sensors, mount for DRO display, etc...
Also it's not very accurate sized, so I'd to glue a felt strip to hold NanoELS in place.
The spindle encoder connected via GX12 connector, because it's planned to be reusable with other ELS. Stepper cables, and USB cable just passed through hole, and secured by ties.

**Comment by Festivejelly** [comment]
How the the outputs of the DRO connected? I have aviation connectors on the back of mine.

**Reply by vegetate7** [reply]
> How the the outputs of the DRO connected? I have aviation connectors on the back of mine.

You'd search for docs specific to your's DRO. They all can be different.
To the NanoELS they connected on AUX ports:
GND    Ax1
A          Ax2
+5V     +5V
B          Ax3
In my setup sensors connected to DRO display  and NanoELS through Y-splitters made of ADUM1200  Dual-channel digital isolator.

**Reply by Festivejelly** [reply]
Thanks for the reply. It turns out my DRO and scales are only accurate to 0.05mm which would make the controller less accurate. So im going to stick with using dial indicators to measure backlash.

**Comment by RGUERI1** [comment]
Nice job. Could you share your code file?

**Reply by vegetate7** [reply]
https://github.com/vegetate7/nanoels_dro
